server/budgettracker.web/Controllers/BudgetApiController.cs

```diff
@@ -21,9 +24,26 @@ public BudgetApiController(IBudgetApi budgetApi)
         }
 
         [HttpPost("create")]
-        public async Task<ApiResponse> CreateBudget(ApiRequest request)
+        public async Task<IActionResult> CreateBudget(ApiRequest request)
         {
-            return await _budgetApi.CreateBudget(request);
+            if(!request.Arguments<CreateBudgetRequestContract>().IsValid())
+            {
+                return BadRequest();
+            }
+
+            try
+            {
+                return Ok(await _budgetApi.CreateBudget(request));    
```

Comment:
Instead of manually returning an object, we should return [JsonResult](https://docs.microsoft.com/en-us/dotnet/api/system.web.mvc.jsonresult?view=aspnet-mvc-5.2) which extends ActionResult. That way, it sets the Content-Type header and all that stuff too. I assume it automatically returns as a 200 response too.

Reply:
Or, I've just been returning the ApiResponse object itself. Either works